internal/stats/aggregate.go

```diff
@@ -0,0 +1,212 @@
+package stats
+
+import (
+	"encoding/json"
+	"fmt"
+	"reflect"
+	"strings"
+
+	"github.com/edulinq/autograder/internal/log"
+	"github.com/edulinq/autograder/internal/util"
+)
+
+type AggregationQuery struct {
+	EnableAggregation bool     `json:"enable-aggregation,omitempty"`
```

Comment:
I'm pretty hesitant about this field.
What if the other fields are non-empty, but this is false?
Can't we check check if the other fields are empty?